lib/order.rb

```diff
@@ -0,0 +1,36 @@
+require 'dotenv/load'
+require_relative 'menu'
+require_relative 'sms'
+
+class Order
+  attr_reader :current_total
+
+  def initialize
+    @current_total = 0
+    @menu = Menu.new
+    @sms = Sms.new
+  end
+
+  def add(list, customer_number)
+    add_up_given(list)
+    fail 'Incorrect total' if incorrect?(list[:total])
+    confirm_order(customer_number)
+  end
+
+  private
+
+  def add_up_given(list)
+    list.each do |dish, amount|
```

Comment:
Have you come across `Enumerable#reduce` or `Enumerable#inject`? Neither are particularly readable (to me, at least!) but either could be used to turn this into a one-liner.

Reply:
I've not been able to figure out `Enumerable#inject` (seems to be the right one as I'd need to pass a block) as I need to be able to filter out one of the key/value pairs passed by the hash. I did however change it to: 
`list.each { |dish, amount| @current_total += @menu.prices[dish].to_f * amount }`